src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -444,5 +460,182 @@ internal static bool ContainsPathComponent(string path, string component, bool i
 
             return false;
         }
+
+        /// <summary>
+        /// Gets a path relative to a directory.
+        /// </summary>
+        public static string GetRelativePath(string directory, string fullPath)
```

Comment:
We've had out function for a long time (many releases)... I am just moving the code into one utilities class.